File: CS/frmPickup.cs
// VBConversions Note: VB project level imports
using System.Data;
using System.Collections.Generic;
using PDSA.Data;
using PDSA.WinForms.Forms;
using PDSA.Security;
using System.Diagnostics;
using PDSA.Common;
using PDSA.WinForms;
using PDSA.WinForms.Security;
using PDSA.PDSACryptography;
using DoorsPOS.Utility;
using Microsoft.VisualBasic;
using System.Collections;
using System;
using DoorsPOS.BusinessLayer;
using System.Windows.Forms;
using System.Drawing;
using System.Linq;
using DoorsPOS.DataLayer;
// End of VB project level imports

using PDSA.DataLayer.DataClasses;
using PDSA.Validation;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Parameters;

namespace DoorsPOS.WinApp
{
	public partial class frmPickup
	{
		public frmPickup()
		{
			InitializeComponent();
		}
		public void btnsave_Click(object sender, EventArgs e)
		{
			
			if (ceamount.Value <= 0)
			{
				MessageBox.Show("Please enter amount picked up.");
				return;
			}
			
			if ((depaydate.EditValue).ToString() == "")
			{
				MessageBox.Show("Please enter amount picked up.");
				return;
			}
			
			int ansint = 7;
			ansint = (int) (Interaction.MsgBox("Are you sure you want to Save the Picked-up Cash?", MsgBoxStyle.YesNo, "Pick-up Cash Save Confirmation"));
			if (ansint == 6)
			{
				try
				{
					//usb printer
					VarsModule.OpenCashdrawer();
					
					//'lpt printer
					//Dim intFileNo As Integer = FreeFile()
					//FileOpen(1, "c:\escapes.txt", OpenMode.Output)
					//PrintLine(1, Chr(27) & "p" & Chr(0) & Chr(25) & Chr(250))
					//FileClose(1)
					//Shell("print /d:com1 c:\escapes.txt", vbNormalFocus)
					
					
					PDSATransaction tranmgr = new PDSATransaction();
					petty_cashManager mgrpay = new petty_cashManager();
					tblreadingManager mgrrdng = new tblreadingManager();
					
					///''''''''''''
					
					mgrpay.Entity.posdate = System.Convert.ToDateTime(depaydate.EditValue);
					mgrpay.Entity.pcamnt = Convert.ToDecimal(ceamount.Value);
					mgrpay.Entity.pcrem = txtremark.Text;
					mgrpay.Entity.cashier = System.Convert.ToString(PDSAAppConfig.CurrentLoginID);
					
					mgrrdng.DataObject.SelectFilter = tblreadingData.SelectFilters.All;
					mgrrdng.DataObject.WhereFilter = tblreadingData.WhereFilters.CashierStatus;
					mgrrdng.Entity.cashier = System.Convert.ToString(PDSAAppConfig.CurrentLoginID);
					mgrrdng.DataObject.Load();
					if (mgrrdng.DataObject.RowsAffected > 0)
					{
						mgrrdng.Entity.pickup = Convert.ToDecimal(ceamount.Value) + mgrrdng.Entity.pickup;
						mgrrdng.DataObject.UpdateFilter = tblreadingData.UpdateFilters.PrimaryKey;
					}
					
					mgrpay.DataObject.TransactionType = PDSA.DataLayer.DataClasses.PDSATransactionType.Insert;
					mgrrdng.DataObject.TransactionType = PDSATransactionType.Update;
					tranmgr.Add(mgrpay.DataObject);
					tranmgr.Add(mgrrdng.DataObject);
					tranmgr.Execute();
					int intgc = 0;
					intgc = mgrpay.Entity.pcid;
					
					xrGetCash xrgetcash = new xrGetCash();
					//posrep.DataSource = sqlDT
					string strcashier = string.Empty;
					strcashier = System.Convert.ToString(PDSAAppConfig.CurrentLoginID);
					//xrgetcash.Cashier.Value = strcashier
					//xrgetcash.DateTme.Value = CDate(depaydate.Text)
					xrgetcash.ptyCashID.Value = intgc;
					xrgetcash.RequestParameters = false;
					xrgetcash.PrintingSystem.ShowMarginsWarning = false;
					xrgetcash.PrintingSystem.StartPrint += ReportOnStartPrint;
					xrgetcash.Print();
					
					
					disable();
					initialvals();
					MessageBox.Show("Pickup Cash successfully saved.");
				}
				catch (PDSAValidationException ex)
				{
					MessageBox.Show(ex.Message);
				}
				catch (Exception ex)
				{
					DisplayException(ex);
				}
				
				this.Close();
			}
		}
		private void ReportOnStartPrint(object sender, DevExpress.XtraPrinting.PrintDocumentEventArgs e)
		{
			e.PrintDocument.PrinterSettings.Copies = (short) 2;
		}
		private void DisplayException(Exception ex)
		{
			//tbException.Text = ex.ToString()
			MessageBox.Show(ex.ToString());
			//MessageBox.Show("Exception Occurred. Check the Exception Tab for More Info.")
		}
		
		public void enable()
		{
			btnnew.Enabled = false;
			btnsave.Enabled = true;
			btncancel.Enabled = true;
			btnclose.Enabled = false;
		}
		public void disable()
		{
			btnnew.Enabled = true;
			btnsave.Enabled = false;
			btncancel.Enabled = false;
			btnclose.Enabled = true;
		}
		public void initialvals()
		{
			ceamount.Value = 0;
			txtremark.Text = string.Empty;
			depaydate.EditValue = DateAndTime.Today;
		}
		
		public void btnnew_Click(object sender, EventArgs e)
		{
			enable();
			initialvals();
		}
		
		public void btncancel_Click(object sender, EventArgs e)
		{
			int ansint = 7;
			ansint = (int) (Interaction.MsgBox("Are you sure you want to cancel?", MsgBoxStyle.YesNo, "Cancel Confirmation"));
			if (ansint == 6)
			{
				disable();
				initialvals();
			}
		}
		
		public void btnclose_Click(object sender, EventArgs e)
		{
			if (btnsave.Enabled == true)
			{
				int ans = 0;
				ans = (int) (Interaction.MsgBox("Pending Additional Cash Transaction not saved. Click Yes to Save or No to Cancel Payment and Close the form.", MsgBoxStyle.YesNo, null));
				if (ans == 7)
				{
					return;
				}
				else
				{
					this.Close();
				}
			}
			else
			{
				this.Close();
			}
		}
		
		public void frmAddtnlCash_FormClosed(object sender, FormClosedEventArgs e)
		{
			this.Dispose();
		}
		
		public void frmPickup_KeyDown(object sender, KeyEventArgs e)
		{
			if (e.KeyCode == Keys.Escape)
			{
				if (btnsave.Enabled == true)
				{
					int ans = 0;
					ans = (int) (Interaction.MsgBox("Pending Pick-up Cash Transaction not saved. Click Yes to Save or No to Cancel Payment and Close the form.", MsgBoxStyle.YesNo, null));
					if (ans == 7)
					{
						return;
					}
					else
					{
						this.Close();
					}
				}
				else
				{
					this.Close();
				}
			}
		}
		
		public void frmPickup_Load(object sender, EventArgs e)
		{
			depaydate.EditValue = DateAndTime.Today;
		}
	}
}

File: frmPickup.vb
Imports PDSA.DataLayer.DataClasses
Imports PDSA.Validation
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Parameters
Public Class frmPickup
    Private Sub btnsave_Click(sender As Object, e As EventArgs) Handles btnsave.Click

        If ceamount.Value <= 0 Then
            MessageBox.Show("Please enter amount picked up.")
            Exit Sub
        End If

        If CStr(depaydate.EditValue) = "" Then
            MessageBox.Show("Please enter amount picked up.")
            Exit Sub
        End If

        Dim ansint As Integer = 7
        ansint = MsgBox("Are you sure you want to Save the Picked-up Cash?", MsgBoxStyle.YesNo, "Pick-up Cash Save Confirmation")
        If ansint = 6 Then
            Try
                'usb printer
                OpenCashdrawer()

                ''lpt printer
                'Dim intFileNo As Integer = FreeFile()
                'FileOpen(1, "c:\escapes.txt", OpenMode.Output)
                'PrintLine(1, Chr(27) & "p" & Chr(0) & Chr(25) & Chr(250))
                'FileClose(1)
                'Shell("print /d:com1 c:\escapes.txt", vbNormalFocus)


                Dim tranmgr As PDSATransaction = New PDSATransaction()
                Dim mgrpay As New petty_cashManager
                Dim mgrrdng As New tblreadingManager()

                '''''''''''''''

                mgrpay.Entity.posdate = CDate(depaydate.EditValue)
                mgrpay.Entity.pcamnt = Convert.ToDecimal(ceamount.Value)
                mgrpay.Entity.pcrem = txtremark.Text
                mgrpay.Entity.cashier = PDSAAppConfig.CurrentLoginID

                mgrrdng.DataObject.SelectFilter = tblreadingData.SelectFilters.All
                mgrrdng.DataObject.WhereFilter = tblreadingData.WhereFilters.CashierStatus
                mgrrdng.Entity.cashier = PDSAAppConfig.CurrentLoginID
                mgrrdng.DataObject.Load()
                If mgrrdng.DataObject.RowsAffected > 0 Then
                    mgrrdng.Entity.pickup = Convert.ToDecimal(ceamount.Value) + mgrrdng.Entity.pickup
                    mgrrdng.DataObject.UpdateFilter = tblreadingData.UpdateFilters.PrimaryKey
                End If

                mgrpay.DataObject.TransactionType = PDSA.DataLayer.DataClasses.PDSATransactionType.Insert
                mgrrdng.DataObject.TransactionType = PDSATransactionType.Update
                tranmgr.Add(mgrpay.DataObject)
                tranmgr.Add(mgrrdng.DataObject)
                tranmgr.Execute()
                Dim intgc As Integer = 0
                intgc = mgrpay.Entity.pcid

                Dim xrgetcash As New xrGetCash()
                'posrep.DataSource = sqlDT
                Dim strcashier As String = String.Empty
                strcashier = PDSAAppConfig.CurrentLoginID
                'xrgetcash.Cashier.Value = strcashier
                'xrgetcash.DateTme.Value = CDate(depaydate.Text)
                xrgetcash.ptyCashID.Value = intgc
                xrgetcash.RequestParameters = False
                xrgetcash.PrintingSystem.ShowMarginsWarning = False
                AddHandler xrgetcash.PrintingSystem.StartPrint, AddressOf ReportOnStartPrint
                xrgetcash.Print()


                disable()
                initialvals()
                MessageBox.Show("Pickup Cash successfully saved.")
            Catch ex As PDSAValidationException
                MessageBox.Show(ex.Message)
            Catch ex As Exception
                DisplayException(ex)
            End Try

            Me.Close()
        End If
    End Sub
Private Sub ReportOnStartPrint(ByVal sender As Object, ByVal e As DevExpress.XtraPrinting.PrintDocumentEventArgs)
        e.PrintDocument.PrinterSettings.Copies = 2
    End Sub
    Private Sub DisplayException(ByVal ex As Exception)
        'tbException.Text = ex.ToString()
        MessageBox.Show(ex.ToString)
        'MessageBox.Show("Exception Occurred. Check the Exception Tab for More Info.")
    End Sub

    Sub enable()
        btnnew.Enabled = False
        btnsave.Enabled = True
        btncancel.Enabled = True
        btnclose.Enabled = False
    End Sub
    Sub disable()
        btnnew.Enabled = True
        btnsave.Enabled = False
        btncancel.Enabled = False
        btnclose.Enabled = True
    End Sub
    Sub initialvals()
        ceamount.Value = 0
        txtremark.Text = String.Empty
        depaydate.EditValue = DateAndTime.Today
    End Sub

    Private Sub btnnew_Click(sender As Object, e As EventArgs) Handles btnnew.Click
        enable()
        initialvals()
    End Sub

    Private Sub btncancel_Click(sender As Object, e As EventArgs) Handles btncancel.Click
        Dim ansint As Integer = 7
        ansint = MsgBox("Are you sure you want to cancel?", MsgBoxStyle.YesNo, "Cancel Confirmation")
        If ansint = 6 Then
            disable()
            initialvals()
        End If
    End Sub

    Private Sub btnclose_Click(sender As Object, e As EventArgs) Handles btnclose.Click
        If btnsave.Enabled = True Then
            Dim ans As Integer = 0
            ans = MsgBox("Pending Additional Cash Transaction not saved. Click Yes to Save or No to Cancel Payment and Close the form.", MsgBoxStyle.YesNo)
            If ans = 7 Then
                Exit Sub
            Else
                Me.Close()
            End If
        Else
            Me.Close()
        End If
    End Sub

    Private Sub frmAddtnlCash_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        Me.Dispose()
    End Sub

    Private Sub frmPickup_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown
        If e.KeyCode = Keys.Escape Then
            If btnsave.Enabled = True Then
                Dim ans As Integer = 0
                ans = MsgBox("Pending Pick-up Cash Transaction not saved. Click Yes to Save or No to Cancel Payment and Close the form.", MsgBoxStyle.YesNo)
                If ans = 7 Then
                    Exit Sub
                Else
                    Me.Close()
                End If
            Else
                Me.Close()
            End If
        End If
    End Sub

    Private Sub frmPickup_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        depaydate.EditValue = DateAndTime.Today
    End Sub
End Class